core/block/detailservice/set_details.go

```diff
@@ -57,6 +59,33 @@ func (s *service) SetWorkspaceDashboardId(ctx session.Context, workspaceId strin
 	return id, err
 }
 
+func (s *service) SetSource(ctx session.Context, objectId string, source []string) error {
+	return cache.Do(s, objectId, func(sb smartblock.SmartBlock) error {
+		st := sb.NewStateCtx(ctx)
+
+		// nolint:errcheck
+		_ = st.Iterate(func(b simple.Block) (isContinue bool) {
```

Comment:
encapsulate this in `simple/dataview` block

Reply:
```
if dv := b.Model().GetDataview(); dv != nil {
				for _, view := range dv.Views {
					view.DefaultTemplateId = ""
					view.DefaultObjectTypeId = ""
				}
				st.Set(b)
				return false
			}
```

Reply:
I have implemented separate method in editor.Dataview for setting source in sets